File: ios/BrunnenZurich/Models/Fountain.swift
import CoreLocation

struct Fountain: Identifiable, Hashable {
    let id: Int
    let name: String
    let location: String?
    let quartier: String?
    let baujahr: Int?
    let brunnenNummer: String?
    let wasserart: String?
    let isTrinkwasser: Bool
    let latitude: Double
    let longitude: Double
    let photoUrl: String?
    let isAbgestellt: Bool

    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
    }

    var clLocation: CLLocation {
        CLLocation(latitude: latitude, longitude: longitude)
    }
}

struct CheckIn: Identifiable, Codable {
    var id: String = UUID().uuidString
    var userId: String
    var fountainId: Int
    var fountainName: String
    var timestamp: Date
    var points: Int
}

struct UserProfile: Identifiable, Codable, Hashable {
    var id: String { uid }
    var uid: String
    var email: String
    var nickname: String
    var avatarUrl: String?
    var points: Int
    var checkInCount: Int
    var friends: [String]
}

File: android/app/src/main/java/app/brunnen/zurich/data/Fountain.kt
package app.brunnen.zurich.data

data class Fountain(
    val id: Int,
    val name: String,
    val location: String?,
    val quartier: String?,
    val baujahr: Int?,
    val brunnenNummer: String?,
    val wasserart: String?,
    val isTrinkwasser: Boolean,
    val latitude: Double,
    val longitude: Double,
    val photoUrl: String?,
    val isAbgestellt: Boolean,
)

data class CheckIn(
    val id: String = "",
    val userId: String = "",
    val fountainId: Int = 0,
    val fountainName: String = "",
    val timestamp: Long = 0L,
    val points: Int = 0,
)

data class UserProfile(
    val uid: String = "",
    val email: String = "",
    val nickname: String = "",
    val avatarUrl: String? = null,
    val points: Long = 0L,
    val checkInCount: Long = 0L,
    val friends: List<String> = emptyList(),
)
